src/metabase/task/creator_sentiment_emails.clj

```diff
@@ -0,0 +1,127 @@
+   [metabase.db :as mdb]
+   [metabase.driver.sql.query-processor :as sql.qp]
+   [metabase.email :as email]
+   [metabase.email.messages :as messages]
+   [metabase.models.setting :as setting]
+   [metabase.public-settings :as public-settings]
+   [metabase.public-settings.premium-features :as premium-features]
+   [metabase.task :as task]
+   [metabase.util.i18n :refer [deferred-tru]]
+   [metabase.util.log :as log]
+   [toucan2.core :as t2]))
+
+(set! *warn-on-reflection* true)
+
+(setting/defsetting ^:private no-surveys
+  (deferred-tru "Enable or disable creator sentiment emails")
+  :type       :boolean
+  :default    true
+  :visibility :internal
+  :audit      :getter)
+
+(defn- fetch-creators
+  [has-whitelabelling?]
+  (t2/query {:select [[:u.email :email]
+                      [:u.date_joined :created_at]
+                      [:u.first_name :first_name]
+                      [[:count [:distinct [:case [:= :d.archived false] :d.id]]] :num_dashboards]
+                      [[:count [:distinct [:case [:and [:= :rc.type "question"] [:= :rc.archived false]] :rc.id]]] :num_questions]
+                      [[:count [:distinct [:case [:and [:= :rc.type "model"] [:= :rc.archived false]] :rc.id]]] :num_models]]
+             :from [[:core_user :u]]
+             :join [[:report_card :rc] [:= :rc.creator_id :u.id]
+                    [:report_dashboard :d] [:= :d.creator_id :u.id]]
+             :where [:and
+                     [:>= :rc.created_at (sql.qp/add-interval-honeysql-form (mdb/db-type) :%now -2 :month)]
+                     [:>= :d.created_at (sql.qp/add-interval-honeysql-form (mdb/db-type) :%now -2 :month)]
+                     [:= :u.is_active true]
+                     [:= :u.type "personal"]
+                     (when has-whitelabelling? [:= :user.is_superuser true])]
+             :group-by [:u.id]
+             :having [:and
+                      [:>= [:count [:distinct :rc.id]] 10]
+                      [:>= [:count [:distinct [:case [:= :rc.query_type "native"] :rc.id]]] 2]
+                      [:>= [:count [:distinct :d.id]] 1]]}))
+
+(defn- fetch-num-users []
+  (->
+   (t2/query-one {:select [[:%count.* :num_users]]
+                  :from [[:core_user :u]]
+                  :where [:and
+                          [:= :u.is_active true]
+                          [:= :u.type "personal"]]})
+   :num_users))
+
+(defn- fetch-num-dashboards []
+  (->
+   (t2/query-one {:select [[:%count.* :num_dashboards]]
+                  :from [[:report_dashboard :d]]
+                  :where [:= :d.archived false]})
+   :num_dashboards))
+
+(defn- fetch-num-cards []
+  (t2/query-one {:select [[[:count [:distinct [:case [:and [:= :rc.type "question"] [:= :rc.archived false]] :rc.id]]] :num_questions]
+                          [[:count [:distinct [:case [:and [:= :rc.type "model"] [:= :rc.archived false]] :rc.id]]] :num_models]]
+                 :from [[:report_card :rc]]}))
+
+(defn- fetch-plan-info []
```

Comment:
This seems like something which would be useful elsewhere in the app, we run into stuff like this all the time.